app/controllers/kernel_builds_controller.rb

```diff
@@ -13,10 +14,14 @@ def show
   # GET /kernel_builds/new
   def new
     @kernel_build = KernelBuild.new
+    @current_user_kernel_configs ||= KernelConfig.where(user_id: current_user.id).distinct
+    @current_user_kernel_sources ||= KernelSource.where(user_id: current_user.id).distinct
   end
 
   # GET /kernel_builds/1/edit
   def edit
+    @current_user_kernel_configs ||= KernelConfig.where(user_id: current_user.id).distinct
+    @current_user_kernel_sources ||= KernelSource.where(user_id: current_user.id).distinct
```

Comment:
Same deal as above.  This may be a good spot to refactor into a `before_action` to load the configs and sources as well, so we don't have to repeat this code.